test/e2e/framework/service_util.go

```diff
@@ -806,6 +869,108 @@ func (j *ServiceTestJig) waitForPodsReady(namespace string, pods []string) error
 	return nil
 }
 
+// testEndpointReachability tests reachability to endpoints(IP, ServiceName..) and ports. Test request is initiated from specified execPod.
+func testEndpointReachability(endpoint string, port int32, protocol v1.Protocol, execPod *v1.Pod) {
+	ep := net.JoinHostPort(endpoint, strconv.Itoa(int(port)))
+	cmd := ""
+	switch protocol {
+	case v1.ProtocolTCP:
+		cmd = fmt.Sprintf("curl -q -s --connect-timeout 2 http://%s", ep)
+	case v1.ProtocolUDP:
+		// TODO: consider Windows compatible command
```

Comment:
This has been introduced considering service with same port but different protocol.

Reply:
I think Andrew's point is that this could cause a false "success" for non-TCP services, couldn't it?

Reply:
yeah, the false positive was the main concern.

Reply:
>this could cause a false "success" for non-TCP services

Changing the TCP and UDP connection test command (~~curl~~) can address this issue
i.e. 
```
case v1.ProtocolTCP:
    cmd = fmt.Sprintf("nc -z -t -G 2 %s %v", endpoint, port)
case v1.ProtocolUDP:
    cmd = fmt.Sprintf("nc -z -u -w 2 %s %v", endpoint, port)
```
Does it make sense?

Reply:
Not sure `nc` is available on Windows agnhost containers. @bclau do you know?

Reply:
There is. The Windows agnhost image is based on the busybox image, which has nc and other Linux common utils and tools.

Reply:
Updated command to use `nc`